scene/main/scene_tree.cpp

```diff
@@ -1217,20 +1217,75 @@ void SceneTree::get_nodes_in_group(const StringName &p_group, List<Node *> *p_li
 void SceneTree::_flush_delete_queue() {
 	_THREAD_SAFE_METHOD_
 
-	while (delete_queue.size()) {
-		Object *obj = ObjectDB::get_instance(delete_queue.front()->get());
+	// Sorting the delete queue by child count (in respect to their parent)
+	// is an optimization because nodes benefit immensely from being deleted
+	// in reverse order to their child count. This is partly due to ordered_remove(), and partly
+	// due to notifications being sent to children that are moved, further in the child list.
+	struct ObjectIDComparator {
+		_FORCE_INLINE_ bool operator()(const DeleteQueueElement &p, const DeleteQueueElement &q) const {
+			return (p.child_list_id > q.child_list_id);
+		}
+	};
+
+	delete_queue.sort_custom<ObjectIDComparator>();
+
+	for (unsigned int e = 0; e < delete_queue.size(); e++) {
```

Comment:
```suggestion
	for (uint32_t e = 0; e < delete_queue.size(); e++) {
```
Is there a reason to use `unsigned int` over `uint32_t`? I think `uint32_t` is more common in the codebase, and `size()` returns a `uint32_t`:
![image](https://user-images.githubusercontent.com/50304111/175981929-095e6aaf-bc3f-47e9-8ff8-dbce453b43cf.png)